// plugin-welcome/src/api/models/EntMedicalEquipment.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntMedicalEquipmentEdges,
    EntMedicalEquipmentEdgesFromJSON,
    EntMedicalEquipmentEdgesFromJSONTyped,
    EntMedicalEquipmentEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalEquipment
 */
export interface EntMedicalEquipment {
    /**
     * 
     * @type {EntMedicalEquipmentEdges}
     * @memberof EntMedicalEquipment
     */
    edges?: EntMedicalEquipmentEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntMedicalEquipment
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntMedicalEquipment
     */
    name?: string;
}

export function EntMedicalEquipmentFromJSON(json: any): EntMedicalEquipment {
    return EntMedicalEquipmentFromJSONTyped(json, false);
}

export function EntMedicalEquipmentFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalEquipment {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntMedicalEquipmentEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntMedicalEquipmentToJSON(value?: EntMedicalEquipment | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntMedicalEquipmentEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
