// plugin-welcome/src/api/models/EntMedicalType.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntMedicalTypeEdges,
    EntMedicalTypeEdgesFromJSON,
    EntMedicalTypeEdgesFromJSONTyped,
    EntMedicalTypeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalType
 */
export interface EntMedicalType {
    /**
     * 
     * @type {EntMedicalTypeEdges}
     * @memberof EntMedicalType
     */
    edges?: EntMedicalTypeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntMedicalType
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntMedicalType
     */
    name?: string;
}

export function EntMedicalTypeFromJSON(json: any): EntMedicalType {
    return EntMedicalTypeFromJSONTyped(json, false);
}

export function EntMedicalTypeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalType {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntMedicalTypeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntMedicalTypeToJSON(value?: EntMedicalType | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntMedicalTypeEdgesToJSON(value.edges),
        'id': value.id,
        'name': value.name,
    };
}
