// plugin-welcome/src/api/models/EntMedicalTypeEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntSystemequipment,
    EntSystemequipmentFromJSON,
    EntSystemequipmentFromJSONTyped,
    EntSystemequipmentToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntMedicalTypeEdges
 */
export interface EntMedicalTypeEdges {
    /**
     * Systemequipment holds the value of the systemequipment edge.
     * @type {Array<EntSystemequipment>}
     * @memberof EntMedicalTypeEdges
     */
    systemequipment?: Array<EntSystemequipment>;
}

export function EntMedicalTypeEdgesFromJSON(json: any): EntMedicalTypeEdges {
    return EntMedicalTypeEdgesFromJSONTyped(json, false);
}

export function EntMedicalTypeEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntMedicalTypeEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'systemequipment': !exists(json, 'systemequipment') ? undefined : ((json['systemequipment'] as Array<any>).map(EntSystemequipmentFromJSON)),
    };
}

export function EntMedicalTypeEdgesToJSON(value?: EntMedicalTypeEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'systemequipment': value.systemequipment === undefined ? undefined : ((value.systemequipment as Array<any>).map(EntSystemequipmentToJSON)),
    };
}
