// plugin-welcome/src/api/models/EntPhysician.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPhysicianEdges,
    EntPhysicianEdgesFromJSON,
    EntPhysicianEdgesFromJSONTyped,
    EntPhysicianEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPhysician
 */
export interface EntPhysician {
    /**
     * 
     * @type {EntPhysicianEdges}
     * @memberof EntPhysician
     */
    edges?: EntPhysicianEdges;
    /**
     * Email holds the value of the "email" field.
     * @type {string}
     * @memberof EntPhysician
     */
    email?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPhysician
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntPhysician
     */
    name?: string;
}

export function EntPhysicianFromJSON(json: any): EntPhysician {
    return EntPhysicianFromJSONTyped(json, false);
}

export function EntPhysicianFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPhysician {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntPhysicianEdgesFromJSON(json['edges']),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}

export function EntPhysicianToJSON(value?: EntPhysician | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntPhysicianEdgesToJSON(value.edges),
        'email': value.email,
        'id': value.id,
        'name': value.name,
    };
}
